src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/PackageBuilder.cs

```diff
@@ -865,13 +867,63 @@ internal static string ResolvePackagePath(string searchDirectory, string searchP
             {
                 // If the search does not contain wild cards, and the target path shares the same extension, copy it
                 // e.g. <file src="ie\css\style.css" target="Content\css\ie.css" /> --> Content\css\ie.css
-                return targetPath;
+                packagePath = targetPath;
+                targetPath = null;
             }
             else
             {
                 packagePath = Path.GetFileName(fullPath);
             }
-            return Path.Combine(targetPath ?? String.Empty, packagePath);
+
+            string path = Path.Combine(targetPath ?? string.Empty, packagePath);
+
+            // Translate the tfm alias to its actual framework if necessary.
+            foreach (string knownFolder in PackagingConstants.Folders.Known)
+            {
+                string folderPrefix = knownFolder + Path.DirectorySeparatorChar;
+                if (path.Length > folderPrefix.Length &&
+                    path.StartsWith(folderPrefix, StringComparison.OrdinalIgnoreCase))
+                {
+                    string frameworkPart = path.Substring(folderPrefix.Length);
+                    string aliasString = Path.GetDirectoryName(frameworkPart).Split(Path.DirectorySeparatorChar).First();
+
+                    if (string.IsNullOrEmpty(aliasString))
+                    {
+                        break;
+                    }
+
+                    string frameworkString = null;
+
+                    NuGetFramework framework = null;
+
+                    if (AliasFolderNameMapping != null && AliasFolderNameMapping.TryGetValue(aliasString, out frameworkString))
+                    {
+                        path = folderPrefix + frameworkString + Path.DirectorySeparatorChar + path.Substring(folderPrefix.Length + aliasString.Length + 1);
+                        framework = NuGetFramework.Parse(frameworkString);
+                    }
+                    else
+                    {
+                        try
+                        {
+                            framework = NuGetFramework.Parse(aliasString);
+                        }
+                        catch
+                        {
+                        }
+                    }
+
+
+                    if (framework != null && framework.HasPlatform && framework.PlatformVersion == FrameworkConstants.EmptyVersion)
+                    {
+                        throw new PackagingException(NuGetLogCode.NU1012, string.Format(CultureInfo.CurrentCulture, NuGetResources.InvalidPlatformVersion, framework.GetShortFolderName()));
```

Comment:
From a usability perspective I'm wondering if the warnings would be more useful if they called out the alias instead :) 

Just something to consider for later (potentially, you can ofc say the current impl is better)